bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/util/ValueUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.matter.internal.util;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.types.Type;
+
+/**
+ * @author Dan Cunningham - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ValueUtils {
+    private static final BigDecimal TEMPERATURE_MULTIPLIER = new BigDecimal(100);
+
+    /**
+     * Converts a ZigBee 8 bit level as used in Level Control cluster and others to a percentage
+     *
+     * @param level an integer between 0 and 254
+     * @return the scaled {@link PercentType}
+     */
+    public static PercentType levelToPercent(int level) {
+        int result = (int) Math.round(level * 100.0 / 254.0);
+        return level == 0 ? PercentType.ZERO : new PercentType(Math.max(result, 1));
+    }
+
+    /**
+     * Converts a {@link PercentType} to an 8 bit level scaled between 0 and 254
+     *
+     * @param percent the {@link PercentType} to convert
+     * @return a scaled value between 0 and 254
+     */
+
+    public static int percentToLevel(PercentType percent) {
+        return (int) (percent.floatValue() * 254.0f / 100.0f + 0.5f);
+    }
+
+    /**
+     * Converts a {@link Command} to a ZigBee / Matter temperature integer
+     *
+     * @param type the {@link Type} to convert
+     * @return the {@link Type} or null if the conversion was not possible
+     */
+    public static @Nullable Integer temperatureToValue(Type type) {
+        BigDecimal value = null;
+        if (type instanceof QuantityType<?> quantity) {
+            if (quantity.getUnit() == SIUnits.CELSIUS) {
+                value = quantity.toBigDecimal();
+            } else if (quantity.getUnit() == ImperialUnits.FAHRENHEIT) {
+                QuantityType<?> celsius = quantity.toUnit(SIUnits.CELSIUS);
+                if (celsius == null) {
+                    return null;
+                }
+                value = celsius.toBigDecimal();
+            } else {
+                return null;
+            }
```

Comment:
Suggestion: as you have a null check at line 82, you could avoid to return null at these places.